src/docfx/template/TemplateEngine.cs

```diff
@@ -43,16 +46,75 @@ private TemplateEngine(string templateDir, JsonSchema metadataSchema)
                 .ToDictionary(prop => prop.Key, prop => prop.Value.HtmlMetaName);
         }
 
-        public static (JsonSchemaValidator, JsonSchemaTransformer) GetJsonSchema(Schema schema)
+        public static SourceInfo<string> ReadMimeFromFile(string pathToDocset, string filePath)
+        {
+            SourceInfo<string> mime = default;
+
+            if (filePath.EndsWith(".json", PathUtility.PathComparison))
+            {
+                if (File.Exists(filePath))
+                {
+                    using (var reader = new StreamReader(filePath))
+                    {
+                        mime = JsonUtility.ReadMime(reader, pathToDocset);
+                    }
+                }
+            }
+            else if (filePath.EndsWith(".yml", PathUtility.PathComparison))
+            {
+                if (File.Exists(filePath))
+                {
+                    using (var reader = new StreamReader(filePath))
+                    {
+                        mime = new SourceInfo<string>(YamlUtility.ReadMime(reader), new SourceInfo(pathToDocset, 1, 1));
+                    }
+                }
+            }
+
+            return mime;
+        }
+
+        public static string GetSchemaName(string mime)
+        {
+            if (mime != null && s_schemas.TryGetValue(mime, out var schema))
+            {
+                return schema;
+            }
+
+            return default;
+        }
+
+        public static bool IsData(string mime)
+        {
+            // todo: get `isData` from template JINT script name
+            if (mime != null && s_schemas.TryGetValue(mime, out var schema))
+            {
+                return string.Equals(schema, "ContextObject", StringComparison.OrdinalIgnoreCase) || string.Equals(schema, "TestData", StringComparison.OrdinalIgnoreCase);
+            }
+
+            return false;
+        }
+
+        public static bool Is(string mime, Type type)
```

Comment:
consider `IsLandingData(string mime)` since that is the only one we need to check.